docarray/helper.py

```diff
@@ -443,3 +443,59 @@ def filter_dict(d: Dict) -> Dict:
     :return: filtered dict
     """
     return dict(filter(lambda item: item[1], d.items()))
+
+
+def _version_check(package: str = None, github_repo: str = None):
+    try:
+        import json
+        import sys
+        import warnings
+        from urllib.request import Request, urlopen
+        import pkg_resources
+        from rich import print
+        from rich.panel import Panel
+        from distutils.version import LooseVersion
+
+        if not package:
+            package = vars(sys.modules[__name__])['__package__']
+        if not github_repo:
+            github_repo = package
+
+        cur_ver = LooseVersion(pkg_resources.get_distribution(package).version)
+        req = Request(
+            f'https://pypi.python.org/pypi/{package}/json',
+            headers={'User-Agent': 'Mozilla/5.0'},
+        )
+        with urlopen(
+            req, timeout=5
+        ) as resp:  # 'with' is important to close the resource after use
+            j = json.load(resp)
+            releases = j.get('releases', {})
+            latest_release_ver = list(
+                sorted(LooseVersion(v) for v in releases.keys() if '.dev' not in v)
+            )[-1]
+            if cur_ver < latest_release_ver:
+                print(
+                    Panel(
+                        f'You are using [b]{package} {cur_ver}[/b], but [bold green]{latest_release_ver}[/] is available. '
+                        f'You may upgrade it via [b]pip install -U {package}[/b]. [link=https://github.com/jina-ai/{github_repo}/blob/main/CHANGELOG.md]Read Changelog here[/link].',
+                        title=':new: New version available!',
+                        width=50,
+                    )
+                )
+    except Exception:
+        # no network, too slow, PyPi is down
+        pass
+
+
+def is_latest_version(package: str = None, github_repo: str = None) -> None:
+    """Check if there is a latest version from Pypi, set env `NO_VERSION_CHECK` to disable it.
+
+    :param package: package name if none auto-detected
+    :param github_repo: repo name that contains CHANGELOG if none then the same as package name
+    """
+    import threading
```

Comment:
please do not use `threading`. This caused the deadlock on core, use multiprocessing

Reply:
deadlock in the core is caused by JinaLogger, not this.